### PR TITLE
added new datapoints and pack() and unpack() functions

### PR DESCRIPTION
**Pull request by mestafin**
Is this better?

I messed up the previous PR completely

### CODE REVIEW

File: knx/dpt/formats.go [hunk 1]
```diff
@@ -73,3 +75,17 @@ func unpackF16(data []byte, f *float32) error {
 	*f = 0.01 * float32(m) * float32(uint(1)<<e)
 	return nil
 }
+
+func packInt(i int) []byte {
```
Comment (vapourismo) [inline knx/dpt/formats.go]: This packs a `uint8`/`byte`, the name and parameter type are misleading. I suggest name `packU8` and `uint8` as parameter.

File: knx/dpt/formats.go [hunk 2]
```diff
@@ -73,3 +75,17 @@ func unpackF16(data []byte, f *float32) error {
...
+	data := []byte{0, 0}
+	data[1] = byte(i)
+	return data
```
Comment (vapourismo) [inline knx/dpt/formats.go]: Why not this:

```go
return []byte{0, i}
```

(Assuming you change `i` to `uint8`)

File: knx/dpt/formats.go [hunk 3]
```diff
@@ -73,3 +75,17 @@ func unpackF16(data []byte, f *float32) error {
...
+}
+
+func unpackInt(data []byte, i *int) error {
```
Comment (vapourismo) [inline knx/dpt/formats.go]: Same goes for this: the name and parameter type are misleading.

File: knx/dpt/types.go [hunk 4]
```diff
@@ -8,12 +8,13 @@ import (
...
 
 // A DatapointValue is a value of a datapoint.
+
 type DatapointValue interface {
 	Pack() []byte
 	Unpack(data []byte) error
 }
 
-// Switch is DPT 1.001.
+// Switch DPT 1.001.
```
Comment (vapourismo) [inline knx/dpt/types.go]: Why?

File: knx/dpt/types.go [hunk 5]
```diff
@@ -31,24 +32,106 @@ func (sw Switch) String() string {
...
+// String generates a string representation.
+func (ss StartStop) String() string {
+	if ss {
+		return "Start"
+	}
+	return "Stop"
+}
+
+// DPT 5.001  (0% - 100%)
+type Scaling int
```
Comment (vapourismo) [inline knx/dpt/types.go]: I really dislike `Scaling` as an `int`. Something that is a percentage value should generally be represented as a fraction. Now, since we don't really have fractions in Go, I believe we should pick `float` instead.

File: knx/dpt/types.go [hunk 6]
```diff
@@ -31,24 +32,106 @@ func (sw Switch) String() string {
...
+// String generates a string representation.
+func (oc OpenClose) String() string {
+	if oc {
+		return "Close"
+	}
+	return "Open"
+}
+
+// DPT 1.010 StartStop
+type StartStop bool
```
Comment (vapourismo) [inline knx/dpt/types.go]: Change that to `type StartStop Switch` and you don't have to implement the `Pack`/`Unpack` method.

**Comment by mestafin** [comment]
Let me try again to squash the commits

**Comment by vapourismo** [comment]
Some hints for your next PR:
- Look at the inline review comments
- Squash commits (with `git rebase -i origin/master`)
- Try not to modify things that are unrelated to your set of changes (i.e. cosmetics)

**Comment by vapourismo** [comment]
This is the final set of changes:
https://github.com/vapourismo/knx-go/compare/981d3abc782fbd370f5e3e9b056d575a7c6bece0...master?expand=1
